htmltest/check-link.go

```diff
@@ -280,6 +275,39 @@ func (hT *HTMLTest) checkExternal(ref *htmldoc.Reference) {
 	// TODO check a hash id exists in external page if present in reference (URL.Fragment)
 }
 
+var hostChannelsLock sync.Mutex // Lock used to prevent concurrent updates to hostChannels
```

Comment:
I initially added this as a member of `HTMLTest` but that in turn required `setRedirectLimitCheck()` to be modified by passing a reference instead of a value; this broke redirection tests by increasing the number of issues found from `1` to `2`.

Making this a global is slightly suboptimal, but seemed to be a reasonable compromise. Happy to move it back into `HTMLTest` if that's a better spot for it to live.